torch/optim/sparse_adam.py

```diff
@@ -97,10 +97,37 @@ def step(self, closure=None):
 
         return loss
 
-SparseAdam.__doc__ = r"""Implements lazy version of Adam algorithm suitable for sparse tensors.
+SparseAdam.__doc__ = r"""SparseAdam implements a masked version of the Adam algorithm
+    suitable for sparse gradients. Please note that SparseAdam is NOT equivalent to
+    Adam "but for sparse gradients", and we understand this may be confusing.
+
+    This is a semantically different optimizer where only moments that are non-zero
+    values in the sparse Tensor of gradients get updated, and only those portions of
+    the gradient get applied to the parameters. This is an optimization, since the
+    values not materialized in the sparse Tensor of gradients will be masked out and
+    skipped. Naturally, the more 0s that are masked, the more performant the
+    optimization. We implement this optimization by using sparse layout Tensors to
```

Comment:
We use sparse layout Tensors to optimize this approximation, which means the more gradients that are masked by being set to 0s, the more performant the optimization.